discovery-infra/install_cluster.py

```diff
@@ -85,60 +85,131 @@ def wait_till_installed(client, cluster, timeout=60 * 60 * 2):
 # 2. Running install cluster api
 # 3. Waiting till all nodes are in installing status
 # 4. Downloads kubeconfig for future usage
-def run_install_flow(client, cluster_id, kubeconfig_path, pull_secret,
-                     tf=None, kube_client=None, cluster_deployment=None):
+def run_install_flow(
```

Comment:
yes, that's a much better approach :)
Maybe in the future we'll have two classes that implements the same interface, one for REST and one for kube-api

Reply:
Requires a huge refactoring but I couldn't agree more